File: java/src/eu/semaine/datatypes/xml/BML.java
/**
 * Copyright (C) 2008 DFKI GmbH. All rights reserved.
 * Use is subject to license terms -- see license.txt.
 */
package eu.semaine.datatypes.xml;

/**
 * @author marc
 *
 */
public class BML 
{
    public static final String version = "1.0";
    public static final String namespaceURI = "http://www.mindmakers.org/projects/BML";

    // Elements
    public static final String E_BML = "bml";
	public static final String ROOT_TAGNAME = E_BML;
	public static final String E_SPEECH = "speech";
	public static final String E_TEXT = "text";
	public static final String E_LANGUAGE = "language";
	public static final String E_SYNC = "sync";
	public static final String E_DESCRIPTION = "description";
	public static final String E_HEAD = "head";
	public static final String E_GAZE = "gaze";
	public static final String E_GESTURE = "gesture";
	public static final String E_LOCOMOTION = "locomotion";
	public static final String E_FACE = "face";
	public static final String E_LIPS = "lips";
	public static final String E_BACKCHANNEL = "backchannel";
	
	// Attributes
	public static final String A_ID = "id";
	public static final String A_START = "start";
	public static final String A_END = "end";
	public static final String A_TYPE = "type";

}

File: c++/src/semaine/datatypes/xml/BML.cpp
/*
 *  BML.cpp
 *  semaine
 *
 *  Created by Marc Schröder on 06.11.08.
 *  Copyright 2008 DFKI GmbH. All rights reserved.
 *
 */

#include "BML.h"

namespace semaine {
namespace datatypes {
namespace xml {

const std::string BML::version = "1.0";
const std::string BML::namespaceURI = "http://www.mindmakers.org/projects/BML";

	// Elements
const std::string BML::E_BML = "bml";
const std::string BML::E_ROOT_TAGNAME = "bml";
const std::string BML::E_SPEECH = "speech";
const std::string BML::E_TEXT = "text";
const std::string BML::E_SYNC = "sync";
const std::string BML::E_DESCRIPTION = "description";
const std::string BML::E_HEAD = "head";
const std::string BML::E_GAZE = "gaze";
const std::string BML::E_GESTURE = "gesture";
const std::string BML::E_LOCOMOTION = "locomotion";
const std::string BML::E_FACE = "face";
const std::string BML::E_LIPS = "lips";

	// Attributes
const std::string BML::A_ID = "id";
const std::string BML::A_START = "start";
const std::string BML::A_END = "end";
const std::string BML::A_TYPE = "type";

// Values
// Values of <head type="...">:
const std::string BML::V_NOD = "NOD";
const std::string BML::V_SHAKE = "SHAKE";
const std::string BML::V_TILTRIGHT = "TILT-RIGHT";
const std::string BML::V_TILTLEFT = "TILT-LEFT";
const std::string BML::V_APPROACH = "APPROACH";
const std::string BML::V_RETRACT = "RETRACT";

} // namespace xml
} // namespace datatypes
} // namespace semaine


